[Chiang Qin Kang] iP

chiangqinkang opened the pull request:
# NotAGPT

[**NotAGPT**](https://github.com/chiangqinkang/ip)

NotAGPT helps store tasks, to remind you when they're due. It's

- Test based
- _Easy to use_
- **Lightweight** 😉
All you need to do is

1. Make sure you have Java 17 installed
2. Run the JAR

- If not, download it from the Java website
- If it does not work, make sure your version is Java 17.
- You can do this with `java -version` in your command line.

Features

- [ ] Stores tasks
- [ ] Stores deadlines
- [ ] Saves tasks

> [!NOTE]
> This is a CLI application.

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}```

chiangqinkang committed:
Level-0: Rename, Greet, Exit

chiangqinkang committed:
Level 0

chiangqinkang committed:
Level-3

chiangqinkang committed:
Level-4 Event added

chiangqinkang committed:
Level-4 more

chiangqinkang committed:
Completed level-4, level-5 and level-6

chiangqinkang committed:
Level-7

chiangqinkang committed:
Level-7

chiangqinkang committed:
Merge branch 'master' into branch-level-7

chiangqinkang committed:
Merge pull request #1 from chiangqinkang/branch-level-7

Level-7

chiangqinkang committed:
Level-8

chiangqinkang committed:
Level-8

chiangqinkang committed:
Merge branch 'master' of https://github.com/chiangqinkang/ip

chiangqinkang committed:
A-MoreOOP

chiangqinkang committed:
Merge pull request #2 from chiangqinkang/branch-level-8

Level-8

chiangqinkang committed:
Merge pull request #3 from chiangqinkang/A-MoreOOP

A More-OOP

chiangqinkang committed:
A-Packages

chiangqinkang committed:
A-Packages

chiangqinkang committed:
A-Packages

chiangqinkang committed:
Merge branch 'master' into branch-A-Packages

chiangqinkang committed:
Merge pull request #4 from chiangqinkang/branch-A-Packages

A-Packages

chiangqinkang committed:
A-JUnit

chiangqinkang committed:
A-JUnit

chiangqinkang committed:
A-JUnit

chiangqinkang committed:
A-Junit

chiangqinkang committed:
A-Junit

chiangqinkang committed:
A-Jar

chiangqinkang committed:
Merge pull request #5 from chiangqinkang/branch-A-Jar

A-Jar

chiangqinkang committed:
Merge branch 'master' into branch-A-JUnit

chiangqinkang committed:
Merge pull request #6 from chiangqinkang/branch-A-JUnit

Branch a j unit

chiangqinkang committed:
Merge branch 'master' into branch-A-Gradle

chiangqinkang committed:
Merge pull request #7 from chiangqinkang/branch-A-Gradle

A-JUnit

chiangqinkang committed:
Add documentation

chiangqinkang committed:
Refactor for coding standard

chiangqinkang committed:
Add find feature

chiangqinkang committed:
Merge branch 'master' into branch-Level-9

chiangqinkang committed:
Merge pull request #8 from chiangqinkang/branch-Level-9

Add find feature

chiangqinkang committed:
Merge branch 'master' into branch-A-JavaDoc

chiangqinkang committed:
Merge pull request #9 from chiangqinkang/branch-A-JavaDoc

Add documentation

chiangqinkang committed:
Merge branch 'master' into branch-A-CodingStandard

chiangqinkang committed:
Merge pull request #10 from chiangqinkang/branch-A-CodingStandard

Refactor for coding standard

chiangqinkang committed:
Add Find

chiangqinkang committed:
Fix for coding standard

chiangqinkang committed:
Start of GUI

chiangqinkang committed:
Fix chat feature

chiangqinkang committed:
Merge pull request #11 from chiangqinkang/branch-A-GUI

Branch a GUI

michaelyql commented on src/main/java/CommandLine/Parser.java:
```diff
@@ -0,0 +1,106 @@
+     * Helps print the string supplied in the appropriate format
+     * @param s the string supplied
+     */
+    public static void responseHelper(String s) {
```
I like how you wrote a helper function to abstract the functionality of printing a string to the terminal. Do you think the function name could be improved by using a verb like `formatResponse`, instead of just `responseHelper`?

michaelyql commented on src/main/java/CommandLine/Parser.java:
```diff
@@ -0,0 +1,106 @@
+
+    /**
+     * Parses the input supplied by user, adds it to the task list if possible
+     * This method helps to parse inputs, calling the appropriate commands where possible
```
nit: There should be a full stop at the end of the method description, and a line to separate the description and the method parameters. I noticed this in a few other places as well.

michaelyql commented on src/main/java/CommandLine/Parser.java:
```diff
@@ -0,0 +1,106 @@
+     */
+    public static String parse(TaskList taskList, String s) {
+        Line line = new Line();
+        String[] parts = s.split(" ", 2);
```
Do you think we can use a more descriptive name to improve readability, like `inputTokens`?

michaelyql commented on src/main/java/Task/Deadline.java:
```diff
@@ -0,0 +1,88 @@
+     * Returns the deadline formatted as a string
+     * @return the deadline as a string
+     */
+    public String getBy() {
```
Perhaps a more reader-friendly method name could be used here? For example, `getDeadline` instead of `getBy`

michaelyql reviewed (commented):
LGTM. Just a few nits to fix

sam-theman88 commented on src/main/java/Task/Task.java:
```diff
@@ -0,0 +1,73 @@
+        E,
+    }
+    private String taskName;
+    private boolean completed;
```
Might want to use the prefix "is" in the name of this field e.g isCompleted for better readability

sam-theman88 commented on src/main/java/Task/Task.java:
```diff
@@ -0,0 +1,73 @@
+     * Returns a task with given string
+     * @return   the task and its given information as a string
+     */
+    public String readTask() {
```
To be consistent with names of other methods you defined, might want to rename this method to getName() or getTaskName()

sam-theman88 commented on src/main/java/Task/TaskList.java:
```diff
@@ -0,0 +1,297 @@
+package Task;
+
```
Is there a reason for 2 new lines here? I think 1 should suffice

sam-theman88 commented on src/main/java/Task/TaskList.java:
```diff
@@ -0,0 +1,297 @@
+        String description = parts[2];
+        Task task;
+        list();
+        switch (taskType) {
+        case T:
+            task = ToDo.of(description, taskType);
+            saveTasks();
+            break;
+        case D:
+            try {
+                task = Deadline.of(description + " /by " + parts[3], taskType);
+                saveTasks();
+            } catch (TaskCreationException e) {
+                throw new IllegalArgumentException("Invalid task type: " + taskType);
+            }
+            break;
+        case E:
+            try {
+                task = Event.of(description + "/from " + parts[3] + " /to " + parts[4], taskType);
+                saveTasks();
+            } catch (TaskCreationException e) {
+                System.out.println("Error occurred while parsing task");
+                throw new IllegalArgumentException("Invalid task type: " + taskType);
+            }
+            break;
+        default:
+            throw new IllegalArgumentException("Invalid task type: " + taskType);
+        }
```
Nice use of switch statements here! Great job in following the coding standard for the module too with regards to indentation for the different cases!

sam-theman88 reviewed (commented):
Overall, LGTM! Awesome job in following and adhering to the coding standard for this module, just some nitpicks here and there.

chiangqinkang committed:
Change to Deadline handling

chiangqinkang committed:
Merge pull request #12 from chiangqinkang/branch-A-GUI

Change to Deadline handling

chiangqinkang committed:
Add asssertions to Task methods

Assertions were added to these methods, so that it can be verified that
they are being called with the proper arguments. These assertions help
make sure that they are working as intended, but without them, will not
cause issues with the program.

chiangqinkang committed:
Fix issue with code quality

Code quality issues with respect to github comments have been fixed.
This includes changing of variable names, but mostly includes proper
formatting of the java doc comments.

chiangqinkang committed:
Merge pull request #13 from chiangqinkang/branch-A-CodeQuality

Fix issue with code quality

chiangqinkang committed:
Merge branch 'master' into branch-A-Assertions

chiangqinkang committed:
Merge pull request #14 from chiangqinkang/branch-A-Assertions

Add asssertions to Task methods

chiangqinkang committed:
Add detect duplicate feature

If a task is a duplicate, where name and deadlines are the same for
applicable tasks, it will not be added.

chiangqinkang committed:
Merge pull request #15 from chiangqinkang/branch-C-DetectDuplicates

Add detect duplicate feature

chiangqinkang committed:
Changes

chiangqinkang committed:
Fix deadline and event

chiangqinkang committed:
Fix issues with saving

This fixed an issue with saving tasks. The issue with loading tasks from the save file was also fixed.

chiangqinkang committed:
Fix code quality

chiangqinkang committed:
Update README.md

chiangqinkang committed:
Add image of UI

chiangqinkang committed:
Fix build.gradle file

chiangqinkang committed:
Fix test and README

chiangqinkang committed:
Fix typos in README

chiangqinkang committed:
Change GUI to dark mode

chiangqinkang committed:
Merge branch 'master' of https://github.com/chiangqinkang/ip

chiangqinkang committed:
Update README
